ui/snippets/walletMenu/WalletMenuContent.tsx

```diff
@@ -12,15 +12,27 @@ type Props = {
   address?: string;
   disconnect?: () => void;
   isAutoConnectDisabled?: boolean;
+  openModal: () => void;
+  close: () => void;
 };
 
-const WalletMenuContent = ({ address, disconnect, isAutoConnectDisabled }: Props) => {
+const WalletMenuContent = ({ address, disconnect, isAutoConnectDisabled, openModal, close }: Props) => {
   const { themedBackgroundOrange } = useMenuButtonColors();
+  const [ isModalOpening, setIsModalOpening ] = React.useState(false);
 
   const onAddressClick = React.useCallback(() => {
     mixpanel.logEvent(mixpanel.EventTypes.WALLET_ACTION, { Action: 'Address click' });
   }, []);
 
+  const handleOpenModal = React.useCallback(async() => {
+    setIsModalOpening(true);
+    await openModal();
+    setTimeout(() => {
+      setIsModalOpening(false);
+      close();
+    }, 500);
+  }, [ openModal, close ]);
```

Comment:
😵‍💫 what will `close` function do? close the opened modal?

Reply:
What modal will be opened?

The prop names could be more specific and self-descriptive to avoid confusing the reader. Please refine them.